gen/Makefile

```
@@ -0,0 +1,28 @@
+GENEREATORS = chains cctpDomains
+chains_TARGET = ../src/constants/Chains.sol
+cctpDomains_TARGET = ../src/constants/CCTPDomains.sol
+
+TEST_WRAPPERS = BytesParsing
+BytesParsing_BASE_PATH = libraries
+
+fnTestWrapperTarget = ../test/generated/$(1)TestWrapper.sol
+TEST_WRAPPER_TARGETS =\
+  $(foreach wrapper, $(TEST_WRAPPERS), $(call fnTestWrapperTarget,$(wrapper)))
+
+.PHONY: generate $(GENEREATORS)
+
+build: $(GENEREATORS) $(TEST_WRAPPER_TARGETS)
+
+$(GENEREATORS): node_modules
+	npx ts-node $@.ts > $($@_TARGET)
+
+node_modules:
+	npm install
```

this target will only run once, then the directory will go stale.
something like this would be more robust:

```suggestion
node_modules: package-lock.json
	npm ci
```